src/DataValues/DecimalMath.php

```diff
@@ -252,115 +252,62 @@ private function roundDigits( $value, $significantDigits ) {
 			return '+0';
 		}
 
+		$len = strlen( $value );
+
+		// keeping all digits means no rounding
+		if ( $significantDigits >= ( $len -1 ) ) {
```

Comment:
In other words: Is `$value` guaranteed to have a sign character and a decimal point? If not, the length calculation must do something about this, because `"1234"`, `"1.23"` and `"-1.2"` all have the same string length, but three different significant digits.